### PR TITLE
doc: release_notes: Updates from applications

### PR DESCRIPTION
**Pull request by pdunaj**
Add information about nrf_desktop, edge impulse and pelion.

Signed-off-by: Pawel Dunaj <pawel.dunaj@nordicsemi.no>

### CODE REVIEW

File: doc/nrf/releases/release-notes-latest.rst [hunk 1]
```diff
@@ -169,10 +169,39 @@ Zigbee
...
+* Added the example application integrating edge impulse wrapper library with sensor sampling.
+  The code is located in :file:`ncs/nrf/applications/machine_learning`.
+* Sample reference to a public pre-trained model.
+
+Pelion
+------
+
+* Added integration of Pelion device management library into the |NCS|.
+  For more information refer to :ref:`ug_pelion`.
+
```
Comment (carlescufi) [inline doc/nrf/releases/release-notes-latest.rst:196]: Added an example application…

File: doc/nrf/releases/release-notes-latest.rst [hunk 2]
```diff
@@ -169,10 +169,39 @@ Zigbee
   * Reworked the :ref:`NCP sample <zigbee_ncp_sample>` to work with the simple gateway application.
   * Moved the `NCP Host documentation`_ from the `ZBOSS NCP Host`_ package to the same location as the `external ZBOSS development guide and API documentation`_.
 
+nRF Desktop
+-----------
+
+* Fixed boot reports on USB backend.
+* Use common modules from CAF library.
```
Comment (greg-fer) [inline doc/nrf/releases/release-notes-latest.rst:188]:
```suggestion
* Adapted the application to use common modules from the :ref:`lib_caf` library.
```
@emob-nordic , can you add release notes for the CAF-related tasks?

File: doc/nrf/releases/release-notes-latest.rst [hunk 3]
```diff
@@ -169,10 +169,39 @@ Zigbee
...
+* Minor bugfixes.
```
Comment (greg-fer) [inline doc/nrf/releases/release-notes-latest.rst:189]:
```suggestion
* Fixed minor bugs.
```

File: doc/nrf/releases/release-notes-latest.rst (hunk 1 above)
Comment (pdunaj) [reply doc/nrf/releases/release-notes-latest.rst:208]: changed

File: doc/nrf/releases/release-notes-latest.rst [hunk 4]
```diff
@@ -181,10 +181,39 @@ Zigbee
...
 ======
 
-There are no entries for this section yet.
+The following changes are relevant for all device families.
+
+Edge Impulse
+------------
+
+* Added the example application integrating edge impulse wrapper library with sensor sampling.
+  The code is located in :file:`ncs/nrf/applications/machine_learning`.
```
Comment (greg-fer) [inline doc/nrf/releases/release-notes-latest.rst:200]:
```suggestion
* Added the :ref:`nrf_machine_learning_app` application that integrates the Edge Impulse wrapper library with sensor sampling.
```
I don't recall providing mention where the code is in previous release notes.

File: doc/nrf/releases/release-notes-latest.rst [hunk 5]
```diff
@@ -181,10 +181,39 @@ Zigbee
...
+* Sample reference to a public pre-trained model.
```
Comment (greg-fer) [inline doc/nrf/releases/release-notes-latest.rst:201]: Not sure what this is about. @MarekPieta ?

File: doc/nrf/releases/release-notes-latest.rst [hunk 6]
```diff
@@ -181,10 +181,39 @@ Zigbee
...
+
+Pelion
+------
+
+* Added integration of Pelion device management library into the |NCS|.
+  For more information refer to :ref:`ug_pelion`.
+
```
Comment (greg-fer) [inline doc/nrf/releases/release-notes-latest.rst:208]: Remove blank line.

File: doc/nrf/releases/release-notes-latest.rst [hunk 7]
```diff
@@ -181,10 +181,39 @@ Zigbee
...
+* Added the example application integrating edge impulse wrapper library with sensor sampling.
+  The code is located in :file:`ncs/nrf/applications/machine_learning`.
+* Sample reference to a public pre-trained model.
+
+Pelion
+------
+
+* Added integration of Pelion device management library into the |NCS|.
+  For more information refer to :ref:`ug_pelion`.
```
Comment (greg-fer) [inline doc/nrf/releases/release-notes-latest.rst:207]:
```suggestion
* Added the integration of Pelion Device Management library, available as one of the external submodule repositories in the |NCS|.
  For more information, see :ref:`ug_pelion`.
```

File: doc/nrf/releases/release-notes-latest.rst [hunk 8]
```diff
@@ -181,10 +181,39 @@ Zigbee
...
+
+* Added an example application utilizing the library.
+  The code is located in :file:`ncs/nrf/applications/pelion_client`.
```
Comment (greg-fer) [inline doc/nrf/releases/release-notes-latest.rst:210]:
```suggestion
* Added the :ref:`pelion_client` application that showcases the usage of Pelion Device Management library.
```

File: doc/nrf/releases/release-notes-latest.rst [hunk 9]
```diff
@@ -181,10 +181,39 @@ Zigbee
...
+
+Common Application Framework (CAF)
+----------------------------------
+
+* Some of the application modules of nRF Desktop application were migrated and reused by other applications.
+  For more information refer to :ref:`lib_caf`.
```
Comment (greg-fer) [inline doc/nrf/releases/release-notes-latest.rst:216]:
```suggestion
* Migrated some of the application modules of nRF Desktop application to :ref:`lib_caf` for reuse by other applications.
```

**Review by greg-fer — CHANGES_REQUESTED** [review]
Some suggestions are marked as outdated due to a force push that happened during the review.

File: doc/nrf/releases/release-notes-latest.rst (hunk 5 above)
Comment (pdunaj) [reply doc/nrf/releases/release-notes-latest.rst:201]: That you don't have to train model yourself as there is a default model added to config.

**Comment by pdunaj** [comment]
@greg-fer applied and squashed

**Review by greg-fer — APPROVED** [review]
@b-gent , ready for merge for me.

**Review by mbolivar-nordic — APPROVED** [review]

**Comment by mbolivar-nordic** [comment]
Needs rebase

**Comment by pdunaj** [comment]
rebased

**Review by shantha-14 — APPROVED** [review]

**Comment by b-gent** [comment]
Looking very good.

**Review by b-gent — APPROVED** [review]

**Comment by b-gent** [comment]
Done in https://github.com/nrfconnect/sdk-nrf/pull/4966

**Pull request closed without merging**